src/AudioToolbox/MusicTrack.cs

```diff
@@ -129,6 +129,20 @@ internal override IntPtr ToUnmanaged ()
 
 	public class MusicEventUserData : MidiRawData {
 		public MusicEventUserData () {}
+
+		internal MusicEventUserData (IntPtr handle)
+		{
+			var buffer = new byte[sizeof(int)];
+			Marshal.Copy (handle, buffer, 0, sizeof(int));
+			int length = BitConverter.ToInt32 (buffer, 0);
+
+			var dataPtr = IntPtr.Add (handle, sizeof (int));
+			buffer = new byte[length];
+			Marshal.Copy (dataPtr, buffer, 0, length);
```

Comment:
Afaik IntPtr has the + operator, so this could be simplified to:

    Marshal.Copy (handle + 4, buffer, 0, length);